examples/resolvers-planetscale/grafbase/resolvers/products/update.ts

```diff
@@ -0,0 +1,94 @@
+import { connect, cast } from "@planetscale/database";
+
+const config = {
+  host: process.env.DATABASE_HOST,
+  username: process.env.DATABASE_USERNAME,
+  password: process.env.DATABASE_PASSWORD,
+};
+
+const conn = connect(config);
+
+export const options = {
+  cast(field, value) {
+    switch (field.name) {
+      case "id": {
+        return String(value);
+      }
+      case "onSale": {
+        return Boolean(value);
+      }
+      default: {
+        return cast(field, value);
+      }
+    }
+  },
+};
+
+export default async function ProductsUpdate(_, args) {
+  const {
+    by: { id, slug },
+    input: { name: newName, slug: newSlug, price: newPrice, onSale: newOnSale },
+  } = args;
+
+  let updateStatement = "UPDATE Products SET ";
+  let params = [];
+
+  if (newName !== undefined) {
+    updateStatement += "name = ?, ";
+    params.push(newName);
+  }
+  if (newSlug !== undefined) {
+    updateStatement += "slug = ?, ";
+    params.push(newSlug);
+  }
+  if (newPrice !== undefined) {
+    updateStatement += "price = ?, ";
+    params.push(newPrice);
+  }
+  if (newOnSale !== undefined) {
+    updateStatement += "onSale = ?, ";
+    params.push(newOnSale);
+  }
+
+  if (params.length === 0) {
+    throw new Error("At least one field to update must be provided.");
+  }
+
+  updateStatement = updateStatement.slice(0, -2);
```

Comment:
Is it worth putting all the `name = ?` bits into an array and doing `.join(", ")` on them?

Spent 30 seconds wondering "what's this slice doing, does it need a comment" - but could just avoid it altogether?  Although not sure if you think it's clearer this way?

Reply:
@obmarg good suggestion. You mean something like this?

```ts
const yo = `UPDATE Products SET ${updateParts.join(", ")} WHERE id = ? OR slug = ?`;
```

Reply:
yeah more or less